sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/Iso8601DateSerializer.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.implementation;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import java.io.IOException;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+/**
+ * Custom serializer to serialize {@link java.util.Date} to Iso8601 standard date format "yyyy-MM-dd'T'hh:mm:ss.SSS'Z'".
+ */
+final class Iso8601DateSerializer extends JsonSerializer<Date> {
```

Comment:
As discussed, these serializers are Jackson specific and may have to go into a separate pluggable module in a follow-up PR. This will allow us to support Gson serializers too.